File: python/0169-e-majority-event.py
## https://leetcode.com/problems/majority-element
from typing import List
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        nums_map = {}

        for n in nums:
            nums_map[n] = nums_map.get(n, 0) + 1

        maxCount = max(nums_map.values())

        for key, value in nums_map.items():
            if maxCount == value:
                return key

        return 0

if __name__ == '__main__':
    print(Solution().majorityElement([1,1,1,1,2,2,2,2,3,4,5,6]))

File: kotlin/0169-e-majority-event.kt
// https://leetcode.com/problems/majority-element
class Solution {
    fun majorityElement(nums: IntArray): Int {
        var candidate: Int? = null
        var count = 0

        for (num in nums) {
            if (count == 0) {
                candidate = num
                count = 1
            } else if (candidate == num) {
                count++
            } else {
                count--
            }
        }

        return candidate ?: 0
    }
}